### PR TITLE
[#5861] iadmin resource commands no longer accept resource names containing whitespace (4-2-stable)

### PR DESCRIPTION
**Pull request by korydraughn**

### CODE REVIEW

File: scripts/irods/test/test_iadmin.py [hunk 1]
```diff
@@ -1603,6 +1603,66 @@ def test_modresc_host_path_warning_messages(self):
...
+        return [
+            '{0} Resc'.format(prefix),
+            '{0}\fResc'.format(prefix),
+            '{0}\nResc'.format(prefix),
+            '{0}\rResc'.format(prefix),
+            '{0}\tResc'.format(prefix),
+            '{0}\vResc'.format(prefix)
+        ]
+
+    def test_creating_a_resource_with_whitespace_is_not_allowed__issue_5861(self):
```
Comment (trel) [inline scripts/irods/test/test_iadmin.py:1616]: didn't we make a `setparentresc` also?

Comment (korydraughn) [reply scripts/irods/test/test_iadmin.py:1616]: That is part of the atomic database operations PR. It is still planned, but the atomic PR has taken a backseat to other things.

**Comment by korydraughn** [comment]
Forgot to mention that this PR includes changes that trim leading/trailing whitespace from valid resource names.

Hopefully there aren't any users who have deployments with resource names containing leading/trailing whitespace.

File: scripts/irods/test/test_iadmin.py (hunk 1 above)
Comment (trel) [reply scripts/irods/test/test_iadmin.py:1616]: ah, yes, of course.

**Comment by trel** [comment]
looks good to me - codacy is just jealous - please # it.

**Comment by korydraughn** [comment]
> looks good to me - codacy is just jealous - please # it.

Running through CI

**Comment by korydraughn** [comment]
passed CI

added #

**Pull request merged**